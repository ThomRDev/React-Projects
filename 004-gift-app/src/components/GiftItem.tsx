import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

interface IGiftItemProps {
    src : string
    title : string
}

const GiftItemEl = styled.div`
  animation: fadeIn 0.5s ease forwards;
  position: relative;
  h2{
    position: absolute;
    bottom:0 ;
    left: 0;
    width: 100%;
    text-align: center;
    color: white;
    margin: 0;
    padding: 1em;
    font-family: monospace;
    background:linear-gradient(rgba(255,255,255,0),rgba(0,0,0,0.6) 70%,black);
  }
`

GiftItemEl.displayName="div"

const GiftItem = ({ src,title }:IGiftItemProps) => {

  const hanldeLoad = () => {

  }

  return (
    <GiftItemEl className='item'>
      <img src={src} alt={title} onLoad={ hanldeLoad } style={{ width:"100%",height:"100%",objectFit:"cover" }}/>
      <h2>{title}</h2>
    </GiftItemEl>
  )
}

GiftItem.propTypes = {
    src : PropTypes.string.isRequired,
    title : PropTypes.string.isRequired
}

export default GiftItem