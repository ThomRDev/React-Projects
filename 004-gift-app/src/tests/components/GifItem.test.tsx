import { shallow } from "enzyme"
import GiftItem from "../../components/GiftItem"

describe(`Tests in ${__filename} file`, () => {
    const props = {
        src : "https://media4.giphy.com/media/l3UceJ4tB12SgFfwc/giphy.gif?câ€¦089fd5n6lchf6g1qiw7irbc9802uiy74nsqucnx4kol0z3&rid=giphy.gif",
        title :"Peru"
    }

    const wrapper = shallow(<GiftItem {...props}  />)
    
    test('Must return component <GifItem /> correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    test('the title of the item must be retun '+props.title, () => { 
        
        const title = wrapper.find("h2").text()
        expect(title).toBe(props.title)
    })
    test('the alt of the img must return '+props.title, () => {
        const img = wrapper.find("img")
        expect(img.prop("alt")).toBe(props.title)
    })
    
    test('must equals src to '+props.src, () => { 
        const img = wrapper.find("img")
        // console.log(img.html());
        // img.props() //me da todos las propiedades en forma de json
        expect(img.prop("src")).toBe(props.src)
        
    })
    test('Props of <GifItem />', () => {
        expect(wrapper.prop("className")?.includes("item")).toBe(true)
    })
})