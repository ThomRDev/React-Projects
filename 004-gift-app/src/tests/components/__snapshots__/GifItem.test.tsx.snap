// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in C:\\Users\\ThomMR\\Desktop\\Learning-React-2022\\004-gift-app\\src\\tests\\components\\GifItem.test.tsx file Must return component <GifItem /> correctly 1`] = `
<div
  className="item"
>
  <img
    alt="Peru"
    onLoad={[Function]}
    src="https://media4.giphy.com/media/l3UceJ4tB12SgFfwc/giphy.gif?câ€¦089fd5n6lchf6g1qiw7irbc9802uiy74nsqucnx4kol0z3&rid=giphy.gif"
    style={
      Object {
        "height": "100%",
        "objectFit": "cover",
        "width": "100%",
      }
    }
  />
  <h2>
    Peru
  </h2>
</div>
`;
